packages/aws-cdk/lib/cli/io-host/cli-io-host.ts

```diff
@@ -153,6 +193,86 @@ export class CliIoHost implements IIoHost {
     this.requireDeployApproval = props.requireDeployApproval ?? RequireApproval.BROADENING;
 
     this.stackProgress = props.stackProgress ?? StackActivityProgress.BAR;
+    this.arguments = props.arguments;
+    this.context = props.context;
+  }
+
+  /**
+   * Asynchronous constuction that needs to be called right after construction.
+   */
+  public async begin() {
+    if (true) {
+      this.commandSpan = await this.asIoHelper().span(CLI_PRIVATE_SPAN.COMMAND).begin({});
+
+      // Cannot collect telemetry without these properties
+      if (this.context && this.arguments !== undefined && canCollectTelemetry(this.context)) {
+        // start the telemetry session
+        await this.bindTelemetrySession(this.arguments, this.context.all);
+
+        // connect ctrl-c to ABORT event
+        process.on('SIGINT', async () => {
+          // Send a special Error 
+          await this.end({
+            name: 'AbortedError',
+          });
+        });
```

Comment:
find the specific error and then get synth event to be abort

Reply:
check if this is necessary -- does node process exit always handle `finally` codes